curvefs/src/mds/metaserverclient/metaserver_client.cpp

```diff
@@ -159,6 +169,15 @@ FSStatusCode MetaserverClient::GetLeader(uint32_t poolId, uint32_t copysetId,
         cntl.set_timeout_ms(options_.rpcTimeoutMs);
         CliService2_Stub stub(&channel_);
         stub.GetLeader(&cntl, &request, &response, nullptr);
+
+        uint32_t maxRetry = options_.rpcRetryTimes;
```

Comment:
The logic of retry is mostly similar, so you can consider reusing it

Reply:
> The logic of retry is mostly similar, so you can consider reusing it

Improved the retry logic, but it is hard to reuse.

Reply:
> The logic of retry is mostly similar, so you can consider reusing it

Reuse it with template.